bin/tx-prover/src/proxy/mod.rs

```diff
@@ -535,6 +551,146 @@ impl ProxyHttp for LoadBalancer {
             self.0.add_available_worker(worker).await;
         }
     }
+
+    // The following methods are a copy of the default implementation defined in the trait, but
+    // with tracing instrumentation.
+    // Pingora calls these methods to handle the request/response lifecycle internally and since
+    // the trait is defined in a different crate, we cannot add the tracing instrumentation there.
+    // We use the default implementation by implementing the method for our specific type, adding
+    // the tracing instrumentation and internally calling `ProxyHttp` methods.
+    // ============================================================================================
+    #[tracing::instrument(name = "proxy:early_request_filter", parent = &ctx.parent_span, skip(_session))]
+    async fn early_request_filter(
+        &self,
+        _session: &mut Session,
+        ctx: &mut Self::CTX,
+    ) -> Result<()> {
+        ProxyHttpDefaultImpl.early_request_filter(_session, &mut ()).await
+    }
+
+    #[tracing::instrument(name = "proxy:connected_to_upstream", parent = &ctx.parent_span, skip(_session, _sock, _fd))]
```

Comment:
Should we not be skipping `all` in these? Or do we want to log context etc?

Reply:
We should keep context here, but I'm removing all other params